### ISSUE TITLE
Escaping curly braces in markdown

### ISSUE THREAD

**Issue by jotak** [quarkiverse/quarkus-roq#286]
In a couple of markdown files I have this sort of things written:
```
Here is a promql query: `mymetric{some_stuff}`
```

The curly braces causes an exception:
```
Error id a03e1d61-889e-474b-b1bc-b631ae14b232-3, io.quarkus.qute.TemplateException: Rendering error in template [roq-gen/posts/2024-04-05-agent_metrics_perf.html] line 47: Key "some_stuff" not found in the template data map with keys [site, now, roqVersion, page, locale] in expression {some_stuff}
```

So it conflicts with the variable injection engine. What are my options to fix that? I see that I can escape those with prepending a backslash, but it's not an ideal solution because if I want my markdown to be reused (readable outside of Roq) I'd need to implement some conversion mechanism. Any idea?

**Comment by ia3andy** [comment]
@mkouba ?

**Comment by ia3andy** [comment]
We could have an option in the fm header to disable qute parsing in the md content

**Comment by jotak** [comment]
@ia3andy that won't help me unfortunately, as I _also_ have some images referred to via qute :-/ 
TBH, if the way to go is to escape (or add a leading space after `{` ), I'm ok with that! Was just curious if they were alternatives.

**Comment by mkouba** [comment]
> We could have an option in the fm header to disable qute parsing in the md content

That would work.

> @ia3andy @melloware that won't help me unfortunately, as I _also_ have some images referred to via qute :-/
> TBH, if the way to go is to escape (or add a leading space after `{` ), I'm ok with that! Was just curious if they were alternatives.

So sometimes you need to escape and sometimes not? But then you can't really reuse the markdown outside of roq, can you?

**Comment by jotak** [comment]
> > @ia3andy @melloware that won't help me unfortunately, as I _also_ have some images referred to via qute :-/
> > TBH, if the way to go is to escape (or add a leading space after `{` ), I'm ok with that! Was just curious if they were alternatives.
> 
> So sometimes you need to escape and sometimes not? But then you can't really reuse the markdown outside of roq, can you?

You're right, or at least, my markdown wouldn't show any image in that case. It's mostly theoretical at the moment for me, I'm only trying roq and figuring out how to integrate that with our blog contributing process... I'm not sure if our contributors are going to install quarkus on their machines to get the previews, but probably we'll ask them to do so.

**Comment by mkouba** [comment]
> > > @ia3andy @melloware that won't help me unfortunately, as I _also_ have some images referred to via qute :-/
> > > TBH, if the way to go is to escape (or add a leading space after `{` ), I'm ok with that! Was just curious if they were alternatives.
> > 
> > 
> > So sometimes you need to escape and sometimes not? But then you can't really reuse the markdown outside of roq, can you?
> 
> You're right, or at least, my markdown wouldn't show any image in that case. It's mostly theoretical at the moment for me, I'm only trying roq and figuring out how to integrate that with our blog contributing process... I'm not sure if our contributors are going to install quarkus on their machines to get the previews, but probably we'll ask them to do so.

Well, in theory you can use a [content filter](https://quarkus.io/guides/qute-reference#content-filters) and escape only some output expressions when the template is loaded, but you'd need a way to identify those expressions...

**Comment by ia3andy** [comment]
Ok using `{|` and `|}` was the solution in my case as showing `qute` examples in docs (rendered with Qute) obviously involves a lot of escaping :)

**Comment by ia3andy** [comment]
I am closing this for now, feel free to reopen
